core/ops_builtin_v8.rs

```diff
@@ -722,12 +722,12 @@ fn op_dispatch_exception(
     return;
   }
 
-  // FIXME(bartlomieju): I'm not sure if this assumption is valid... Maybe when
-  // inspector is polling on pause?
-  if state.inspector().try_borrow().is_ok() {
-    scope.terminate_execution();
-  } else {
+  match state.inspector().try_borrow() {
+    Ok(inspector) if !inspector.has_blocking_sessions() => {
+      scope.terminate_execution();
+    }
     // If the inspector is borrowed at this time, assume an inspector is active.
+    _ => {}
   }
```

Comment:
Above commit https://github.com/denoland/deno/pull/18853/commits/aa63f1bd984327d73f32bc4ba1853e63009ccd3e makes this more sound by making it so that only blocking inspector sessions prevent termination here. And only `deno repl` is a blocking session. @bartlomieju Is this FIXME actually still relevant or no?